fix(share/availability): update availability ErrNotAvailable handling logic

walldiss opened the pull request:
## Overview

`ipldFormat.IsNotFound` could no longer reach Availability layer. It will get converted on getter layer to `share.ErrNotFound`. Update outdated error handling.

Wondertan commented:
Nice catch!

Wondertan reviewed (approved):
Is there a way to test this? Like we forgot to updated the error check and ideally there would be a test that would catch it

walldiss commented:
I've added test for error conversion. From my experience tests like this gets outdated and deleted very quickly since they highly depended on implementattion
